### DISCUSSION TITLE
Customized validation without explicitly state "return true" if it's successful

### DISCUSSION THREAD

**Discussion by 549393092** [quasarframework/quasar#12072] [General - Components / Directives / etc] [answered]
Dear Quasar team,

Just a small question about validation, could you please have a look?
After checking Quasar validation https://quasar.dev/vue-components/input#internal-validation
Please see the customized validation below and let's discuss the condition !== 'A', if without adding "return true", the validation is still successful.
```
check (value) {
  if (condition === 'A') {
    return value.includes('Hello') || 'Field must contain word Hello'
  }
  return true // the validation is successful, shall we add this line explicitly?
}
```
So my question is, is that necessary to add code "return true"? Any official documentation will be appreciate.
I just want to waste one line code 😊.

**Comment by pdanpdan** [comment]
Yes, of course it is required. The default return of a function is undefined

**Reply by 549393092** [reply]
@pdanpdan Thanks! Yes, I have also considered about that. However, after testing that, I found that if without add "return true", it still works. So I was thinking that if Quasar did some change, and want to check with your team.

**Comment by pdanpdan** [comment, marked as answer]
In code the returns are considered like this:
- false or a string => error
- true or undefined/void 0 => ok
- promise => wait for resolve and check as above

So technically undefined/no final return is ok, but for code ease of understanding I would always return on all code branches

**Reply by 549393092** [reply]
@pdanpdan Thanks! It's clear.
